frame/contracts/src/storage.rs

```diff
@@ -352,3 +335,81 @@ impl<T: Config> Deref for DepositAccount<T> {
 		&self.0
 	}
 }
+
+/// DeletionQueue manage the removal of contracts storage that are marked for deletion.
+///
+/// When a contract is deleted by calling `seal_terminate` it becomes inaccessible
+/// immediately, but the deletion of the storage items it has accumulated is performed
+/// later by pulling the contract from the queue in the `on_idle` hook.
+#[derive(Encode, Decode, TypeInfo, MaxEncodedLen, Clone)]
+#[scale_info(skip_type_params(T))]
+pub struct DeletionQueue<T: Config> {
+	/// Monotonic counter used as a key for inserting new deleted contract in the DeletionQueueMap.
+	/// The nonce is incremented after each insertion.
+	insert_nonce: u32,
+	/// The index used to read the next element to be deleted in the DeletionQueueMap.
+	/// The nonce is incremented after each deletion.
+	delete_nonce: u32,
+
+	_phantom: PhantomData<T>,
+}
+
+impl<T: Config> Default for DeletionQueue<T> {
+	fn default() -> Self {
+		Self { insert_nonce: 0, delete_nonce: 0, _phantom: Default::default() }
+	}
+}
```

Comment:
We can derive that. There is [`DefaultNoBound`](https://paritytech.github.io/substrate/master/frame_support/derive.DefaultNoBound.html).